py/dml/c_backend.py

```diff
@@ -1745,13 +1745,14 @@ def generate_init(device, initcode, outprefix):
     out('}\n\n', preindent = -1)
     splitting_point()
 
-def generate_extern_trampoline(exported_name, func):
-    params = (func.cparams if func.independent else
-              [("_obj", TPtr(TNamed("conf_object_t")))] + func.cparams[1:])
-    out("extern %s\n" % (func.rettype.declaration(
-                         "%s(%s)" % (exported_name,
-                                     ", ".join(t.declaration(n)
-                                               for (n, t) in params)))))
+def generate_static_trampoline(func):
+    # static trampolines never need to be generated for independent methods
+    assert not func.independent
+    params = [("_obj", TPtr(TNamed("conf_object_t")))] + func.cparams[1:]
```

Comment:
Perhaps we should also validate that `_obj` is an object of the right class? (same thing in generate_extern_trampoline)

Reply:
Done.